modules/core/src/main/java/org/apache/synapse/MessageContext.java

```diff
@@ -164,6 +164,15 @@
      */
     public Object getEntry(String key);
 
+
+    /**
+     * Get the value of a property set on the message instance or from the local registry
```

Comment:
This is not the local registry space in wso2 products. 
In synapse, we call "local registry" which holds artifacts like local-entries, sequences, etc.